src/toil/fileStores/nonCachingFileStore.py

```diff
@@ -237,17 +245,20 @@ def _getAllJobStates(coordination_dir: str) -> Iterator[Dict[str, str]]:
         :return: dict with keys (jobName,  jobProcessName, jobDir)
         """
         jobStateFiles = []
-        
+
+        assert os.path.exists(coordination_dir), f"{coordination_dir} has vanished unexpectedly!"
+
         # Note that the directory may contain files whose names are not decodable to Unicode.
         # So we need to work in bytes.
         for entry in os.scandir(os.fsencode(coordination_dir)):
             # For each job state file in the coordination directory
             if entry.name.endswith(b'.jobState'):
                 # This is the state of a job
                 jobStateFiles.append(os.fsdecode(entry.path))
-        
+
         for fname in jobStateFiles:
             try:
+                logger.debug('Considering other job state file %s', fname)
```

Comment:
Is this debug message still helpful?

Reply:
Only if we run into the same problem again on another scheduler.